var/spack/repos/builtin/packages/aml/package.py

```diff
@@ -3,66 +3,155 @@
+    dependency for building opencl features.
+    - `+all`: enables all above variants.
+    - `openmp-targets={spir64,}`: Enable interoperability of pointers between
+    intel OpenMP and intel level_zero for spir64 openmp targets.
+    - `hip-platform={amd,nvidia}`: Compile aml with this hip backend
+    implementation. Under the hood, hip calls either amd or cuda functions.
+
+    Examples:
+    ========
+
+    - spack install aml +cuda +hwloc +hip +ze +opencl
+    - spack install aml +cuda +hip hip-platform=nvidia
+    - spack install aml +ze openmp-targets=spir64
+    """
 
     homepage = "https://argo-aml.readthedocs.io/"
-    url = "https://www.mcs.anl.gov/research/projects/argo/downloads/aml-0.1.0.tar.gz"
-    git = "https://github.com/anlsys/aml.git"
+
     maintainers = ['perarnau']
 
     test_requires_compiler = True
 
     tags = ['e4s']
 
-    version('0.1.0', sha256='cc89a8768693f1f11539378b21cdca9f0ce3fc5cb564f9b3e4154a051dcea69b')
-    version('master', branch='master', submodules=True)
-
-    depends_on('numactl')
-
+    # Package sources
+    ###################################
+
+    url = 'https://github.com/anlsys/aml/releases/download/v0.2.0/aml-0.2.0.tar.gz'
+    git = 'https://github.com/anlsys/aml.git'
+
+    # version string is generated from git tags, requires entire repo
+    version('master', branch='master', submodules=True, get_full_repo=True)
+
+    version('0.2.0',
+            sha256='2044a2f3f1d7a19827dd9c0726172b690189b4d3fe938656c4160c022468cc4a')
+    version('0.1.0',
+            sha256='cc89a8768693f1f11539378b21cdca9f0ce3fc5cb564f9b3e4154a051dcea69b',
+            deprecated=True)
+
+    # Variants and configure options management
+    ###########################################
+
+    # This is the function to overload to pass all hwloc flag.
+    def configure_args(self):
+        config_args = []
+        for b in ['opencl', 'hwloc', 'ze', 'hip', 'cuda']:
+            config_args.extend(self.with_or_without(b))
+        if 'openmp-targets=spir64' in self.spec:
+            config_args += ['--with-openmp-flags="-fiopenmp -fopenmp-targets=spir64"']
+        if 'hip-platform=amd' in self.spec:
+            config_args += ['--with-hip-platform=amd']
+        if 'hip-platform=nvidia' in self.spec:
+            config_args += ['--with-hip-platform=nvidia']
+        return config_args
+
+    # Generate possible variants.
+    #############################
+
+    for b in ['opencl', 'hwloc', 'ze', 'hip', 'cuda']:
+        variant(b, default=False,
+                description="Enable features relying on {} backend.".format(b))
+    variant('openmp-targets', values=disjoint_sets(('spir64')),
+            description="Intel OpenMP target architecture setting.")
+    variant('hip-platform', values=disjoint_sets(('amd', 'nvidia')),
+            description="Hip backend platform.")
+
+    # Dependencies management
+    #########################
+
```

Comment:
This fixes the build, but you have to apply this suggestion yourself since you denied access by maintainers like me to the branch of your PR:
```suggestion

    # - libnuma from the numactl package is required to build aml in GitLab CI
    depends_on('numactl')
```

Reply:
The missing `depends_on` was added with commit 7bce013 below. This conversation is resolved therefore.